agents-api/tests/test_tool_calls_evaluator.py

```diff
@@ -0,0 +1,1807 @@
+    )
+
+    assert response.status_code == 201
+    assert acompletion.call_count == 2
+
+
+@test("chat: evaluate agent.doc.create tool call")
+async def _(
+    make_request=make_request,
+    developer_id=test_developer_id,
+    agent=test_agent,
+    mocks=agent_doc_create,
+    dsn=pg_dsn,
+):
+    pool = await create_db_pool(dsn=dsn)
+    session = await create_session(
+        developer_id=developer_id,
+        data=CreateSessionRequest(
+            agent=agent.id,
+            situation="test session about",
+            recall_options={
+                "mode": "text",
+                "num_search_messages": 10,
+                "max_query_length": 1001,
+                "confidence": 0.6,
+                "limit": 5,
+                "mmr_strength": 0.5,
+            },
+        ),
+        connection_pool=pool,
+    )
+
+    (_embed, acompletion) = mocks
+
+    response = make_request(
+        method="POST",
+        url=f"/sessions/{session.id}/chat",
+        json={"messages": [{"role": "user", "content": "hello"}]},
+    )
+
+    assert response.status_code == 201
+    assert acompletion.call_count == 2
+
+
+@test("chat: evaluate agent.doc.delete tool call")
+async def _(
+    make_request=make_request,
+    developer_id=test_developer_id,
+    agent=test_agent,
+    mocks=agent_doc_delete,
+    dsn=pg_dsn,
+):
+    pool = await create_db_pool(dsn=dsn)
+    session = await create_session(
+        developer_id=developer_id,
+        data=CreateSessionRequest(
+            agent=agent.id,
+            situation="test session about",
+            recall_options={
+                "mode": "text",
+                "num_search_messages": 10,
+                "max_query_length": 1001,
+                "confidence": 0.6,
+                "limit": 5,
+                "mmr_strength": 0.5,
+            },
+        ),
+        connection_pool=pool,
+    )
+
+    (_embed, acompletion) = mocks
+
+    response = make_request(
+        method="POST",
+        url=f"/sessions/{session.id}/chat",
+        json={"messages": [{"role": "user", "content": "hello"}]},
+    )
+
+    assert response.status_code == 201
+    assert acompletion.call_count == 2
+
+
+@test("chat: evaluate agent.doc.search tool call")
+async def _(
+    make_request=make_request,
+    developer_id=test_developer_id,
+    agent=test_agent,
+    mocks=agent_doc_search,
+    dsn=pg_dsn,
+):
+    pool = await create_db_pool(dsn=dsn)
+    session = await create_session(
+        developer_id=developer_id,
+        data=CreateSessionRequest(
+            agent=agent.id,
+            situation="test session about",
+            recall_options={
+                "mode": "text",
+                "num_search_messages": 10,
+                "max_query_length": 1001,
+                "confidence": 0.6,
+                "limit": 5,
+                "mmr_strength": 0.5,
+            },
+        ),
+        connection_pool=pool,
+    )
+
+    (_embed, acompletion) = mocks
```

Comment:
maybe something like:

```python
   test_agent = client.agents.create(...,
       tools=[dict(name='search_docs', type='system', resource='agent.docs', op='search')]
    )

    with make_acompletion_from_tools(agent.tools) as (_embed, acompletion):
        # since agent.tools has the search_docs function, this should have called agent.docs.search
        ...
```